openpdf/src/main/java/com/lowagie/text/pdf/PdfStamperImp.java

```diff
@@ -326,20 +326,9 @@ else if (includeFileID) {
         }
         // write the cross-reference table of the body
         body.writeCrossReferenceTable(os, root, info, encryption, fileID, prevxref);
-        if (fullCompression) {
-            os.write(getISOBytes("startxref\n"));
-            os.write(getISOBytes(String.valueOf(body.offset())));
-            os.write(getISOBytes("\n%%EOF\n"));
-        }
-        else {
-            PdfTrailer trailer = new PdfTrailer(body.size(),
-            body.offset(),
-            root,
-            info,
-            encryption,
-            fileID, prevxref);
-            trailer.toPdf(this, os);
-        }
+        os.write(getISOBytes("startxref\n"));
+        os.write(getISOBytes(String.valueOf(body.offset())));
+        os.write(getISOBytes("\n%%EOF\n"));
```

Comment:
I cleaned up xref/trailer writing a bit.

An old-style xref table is followed by a trailer, while a new-style xref stream is not because it includes trailer properties itself. Since the writeCrossReferenceTable method might write either kind of xref, it should take care of the trailer itself, to avoid lots of duplicate code.